File: src/main/java/net/minecraftforge/registries/tags/IReverseTag.java
package net.minecraftforge.registries.tags;

import net.minecraft.tags.TagKey;
import java.util.stream.Stream;

public interface IReverseTag<V> {
    default Stream<TagKey<V>> getTagKeys() {
        throw new RuntimeException("mixin, why didn't you add this");
    }

    default boolean containsTag(TagKey<V> key) {
        throw new RuntimeException("mixin, why didn't you add this");
    }

    default boolean containsTag(ITag<V> tag) {
        throw new RuntimeException("mixin, why didn't you add this");
    }
}

File: src/main/kotlin/net/minecraftforge/registries/tags/IReverseTag.kt
package net.minecraftforge.registries.tags

import net.minecraft.tags.TagKey
import java.util.stream.Stream

interface IReverseTag<V> {
    fun getTagKeys(): Stream<TagKey<V>>
    fun containsTag(key: TagKey<V>): Boolean
    fun containsTag(tag: ITag<V>): Boolean {
        return containsTag(tag.key)
    }
}